### DISCUSSION TITLE
Como construir crawlers em NodeJS

### DISCUSSION THREAD

**Discussion by waldemirflj** [frontendbr/forum#334] [Perguntas e Respostas]
Quais as melhores ferramentas/práticas para se construir crawler, scraping e etc... em NodeJS?

**Comment by vitor-mendrone-deel** [comment]
@waldemirflj Editei o título e corpo da sua questão pra deixar mais simples de ser encontrada, ok?

**Comment by expalmer** [comment]
Estou usando [http://www.nightmarejs.org/](http://www.nightmarejs.org/)

**Comment by rogeralbinoi** [comment]
Cara, eu não sei se isso pode ajudar mas quando eu estava pesquisando sobre isso eu encontrei este post http://www.netinstructions.com/how-to-make-a-simple-web-crawler-in-javascript-and-node-js/ e consegui fazer algumas brincadeiras legais.

**Comment by icaromh** [comment]
Dependendo do que tu precisa tu pode usar alguns prontos, por exemplo o [x-ray](https://github.com/lapwinglabs/x-ray)

**Comment by alanhr** [comment]
EU já usei o YQL da yahoo. 
Nele existe um cara chamado HTML que vc consegue pega informações do site a partir de divs, class e etc.

**Comment by brunoventura** [comment]
@waldemirflj opa meio antigo, mas se ainda der tempo, fiz um crawler de brincadeira para apredender melhor o uso de map reduce com promises (com alguns benchmarks). Case lhe ajude, segue o repo:

https://github.com/brunoventura/brejas-crawler
